core/templates/expressions/expression-evaluator.service.spec.ts

```diff
@@ -139,68 +115,49 @@ describe('Expression evaluator service', function() {
-      [['==', true], ests.ExprWrongNumArgsError],
-      [['+', 'abc', 1], ests.ExprWrongArgTypeError]
-    ].forEach(function(test) {
-      var expression = test[0];
-      var expected = test[1];
-
-      // 'expected' should be either a JavaScript primitive value that would be
-      // the result of evaluation 'expression', or an exception that is
-      // expected to be thrown.
-      // 'expression' is either a string (in which case parsed) or an array
-      // (representing a parse tree).
-      var parsed = (
-        isString(expression) ? eps.parse(expression) : expression);
-      var parsedJson = JSON.stringify(parsed);
-      var failed = false;
-
-      var recordFailure = function(result, exception) {
-        console.error('input     : ' + expression);
-        console.error('parsed    : ' + parsedJson);
-        if (result !== undefined) {
-          console.error('evaluated : ' + result);
-          console.error('expected  : ' + expected);
-        }
-        if (exception !== undefined) {
-          console.error('exception : ' + exception);
-          console.error('expected  : (exception)');
-        }
-        failed = true;
-      };
-
-      try {
-        var evaled = ests.applyFunctionToParseTree(parsed, ENVS, ees.evaluate);
-        if (expected instanceof Error || evaled !== expected) {
-          recordFailure(evaled, undefined);
-        }
-      } catch (e) {
-        if (!(e instanceof expected)) {
-          // Wrong or unexpected exception.
-          recordFailure(undefined, e);
-        }
-      }
-      expect(failed).toBe(false);
+    ]).forEach(([expression, expected]) => {
+      it('should evaluate ' + JSON.stringify(expression) + ' correctly', () => {
+        expect(
+          expressionSyntaxTreeService.applyFunctionToParseTree(
+            expressionParserService.parse(expression), ENVS,
+            (parsed, envs) => expressionEvaluatorService.evaluate(parsed, envs)
+          )).toEqual(expected);
+        expect(expressionEvaluatorService.evaluateExpression(expression, ENVS))
+          .toEqual(expected);
+      });
+    });
+  });
 
-      if (typeof expression !== 'string') {
-        return;
-      }
+  describe('Evaluating invalid expressions', () => {
+    type TestParam = [string, Expr[], Type<ExpressionError>];
+    (<TestParam[]> [
+      ['there are too many args', ['+', 10, 20, 30], ExprWrongNumArgsError],
+      ['there are too few args', ['==', true], ExprWrongNumArgsError],
+      ['an arg has the wrong type', ['+', 'abc', 1], ExprWrongArgTypeError],
+    ]).forEach(([errorReason, expression, errorKind]) => {
+      it('should report an error when ' + errorReason, () => {
+        expect(
+          () => expressionSyntaxTreeService.applyFunctionToParseTree(
+            // This throws "type error". This needs to be suppressed because we
+            // are explicitly testing invalid types.
+            // @ts-ignore
+            expression,
```

Comment:
Please use `@ts-expect-error` here. Also after the `This throws` mention the exact typescript error that is thrown.

Reply:
You mean the code, right? Like `error TS2345`?

Reply:
Nope. The complete error message. For eg

https://github.com/oppia/oppia/blob/a927b0e8b951af8017dfef9d43a70a1d2d881b69/core/templates/services/app.service.spec.ts#L35-L40

Reply:
I've decided to change the signature instead so that this is not an error.